src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBuildWrapper.java

```diff
@@ -408,7 +330,7 @@ public void add(SSHUserPrivateKey key) throws IOException, InterruptedException
          */
         @Override
         public void buildEnvVars(Map<String, String> env) {
-            env.put("SSH_AUTH_SOCK", agent.getSocket());
+            env.putAll(agent.getEnv());
```

Comment:
Probably only this one variable really _needs_ to be passed to nested steps, but `ssh-agent(1)` suggests you just `eval` the env, so why not?